lib/ansible/modules/cloud/google/gcp_container_node_pool.py

```diff
@@ -0,0 +1,765 @@
+#
+# ----------------------------------------------------------------------------
+#
+#     This file is automatically generated by Magic Modules and manual
+#     changes will be clobbered when the file is regenerated.
+#
+#     Please read more about how to change this file at
+#     https://www.github.com/GoogleCloudPlatform/magic-modules
+#
+# ----------------------------------------------------------------------------
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+################################################################################
+# Documentation
+################################################################################
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ["preview"],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: gcp_container_node_pool
+description:
+    - NodePool contains the name and configuration for a cluster's node pool.
+    - Node pools are a set of nodes (i.e. VM's), with a common configuration and specification,
+      under the control of the cluster master. They may have a set of Kubernetes labels
+      applied to them, which may be used to reference them during pod scheduling. They
+      may also be resized up or down, to accommodate the workload.
+short_description: Creates a GCP NodePool
+version_added: 2.6
+author: Google Inc. (@googlecloudplatform)
+requirements:
+    - python >= 2.6
+    - requests >= 2.18.4
+    - google-auth >= 1.3.0
+options:
+    state:
+        description:
+            - Whether the given object should exist in GCP
+        required: true
+        choices: ['present', 'absent']
+        default: 'present'
+    name:
+        description:
+            - The name of the node pool.
+        required: false
+    config:
+        description:
+            - The node configuration of the pool.
+        required: false
+        suboptions:
+            machine_type:
+                description:
+                    - The name of a Google Compute Engine machine type (e.g.
+                    - n1-standard-1).  If unspecified, the default machine type is n1-standard-1.
+                required: false
+            disk_size_gb:
+                description:
+                    - Size of the disk attached to each node, specified in GB. The smallest allowed disk
+                      size is 10GB. If unspecified, the default disk size is 100GB.
+                required: false
+            oauth_scopes:
+                description:
+                    - The set of Google API scopes to be made available on all of the node VMs under the
+                      "default" service account.
+                    - 'The following scopes are recommended, but not required, and by default are not
+                      included:  https://www.googleapis.com/auth/compute is required for mounting persistent
```

Comment:
URLs should be in U() to be formatted properly, so `U(https://www.googleapis.com/auth/compute)`. Same for any of the following links.